File: dart/algorithms/recursion/factorial.dart
// factorial.dart

int factorial(int n) {
  if (n == 0 || n == 1) {
    return 1;
  }
  return n * factorial(n - 1);
}

void main() {
  int n = 5;
  print("Factorial of $n is: ${factorial(n)}");
}

File: go/algorithms/math/factorial.go
// factorial.go

package main

import "fmt"

func factorial(n int) int {
    result := 1

    for i := 1; i <= n; i++ {
        result *= i
    }

    return result
}

func main() {
    n := 5
    result := factorial(n)
    fmt.Printf("Factorial of %d is: %d\n", n, result)
}